src/components/SpiceList.js

```diff
@@ -4,15 +4,69 @@ import SpiceItem from './SpiceItem'
 class SpiceList extends React.Component {
   state = {
     fourStarOnly: false,
-    search: ""
+    search: "",
   }
 
   renderSpices() {
-    return this.props.spices.map(spice => (
+    // Set the initial array of spices to filter to be all spices passed down via props
+    let spicesToFilter = this.props.spices;
+
+    // Filter out for all four-star spices, store them in fourStarSpices array
+    const fourStarSpices = this.props.spices.filter(spice => {
+      return spice.rating >= 4
+    })
+
+    // If the state fourStarOnly is true (meaning that 4 Star Only Checkbox is checked), then we want to filter only fourStarSpices, so we set spicesToFilter to be fourStarSpices
+    if (this.state.fourStarOnly) {
+      spicesToFilter = fourStarSpices
+    }
+
+    // We filter out the spicesToFilter array to match the search term that the user entered, taking into account casing; we store these in a variable called filteredSpices
+    const filteredSpices = spicesToFilter.filter(spice => {
+      if (this.state.search.length === 0) {
+        return true
+      } else {
+        return spice.notes.toLowerCase().includes(this.state.search.toLowerCase())
+      }
+    })
```

Comment:
Nice work on the filtering logic here! It's great to see you breaking down this problem to manageable steps. For the filter based on includes, you can simplify it a bit:

```js
    const filteredSpices = spicesToFilter.filter(spice => {
        return spice.notes.toLowerCase().includes(this.state.search.toLowerCase())
    })
```

Because if `this.state.search` is an empty string, `.includes("")` will always return true.